stdlib/collections/__init__.pyi

```diff
@@ -301,3 +310,13 @@ class ChainMap(MutableMapping[_KT, _VT], Generic[_KT, _VT]):
     def pop(self, key: _KT) -> _VT: ...
     @overload
     def pop(self, key: _KT, default: _VT | _T = ...) -> _VT | _T: ...
+    def copy(self: Self) -> Self: ...
+    @classmethod
+    @overload
+    def fromkeys(cls, iterable: Iterable[_T]) -> ChainMap[_T, Any | None]: ...
+    @classmethod
+    @overload
+    def fromkeys(cls, __iterable: Iterable[_T], __value: None) -> ChainMap[_T, Any | None]: ...
+    @classmethod
+    @overload
+    def fromkeys(cls, __iterable: Iterable[_T], __value: _S) -> ChainMap[_T, _S]: ...
```

Comment:
The implementation returns a subclass of `ChainMap`, if `fromkeys()` is called on a subclass. I'm not sure if there is syntax to describe that. With an instance method, you would use `def foo(self: Self) -> Self`.

Reply:
There is no syntax to describe this without higher-kinded `TypeVar`s; it's the reason why `dict.fromkeys` is spelt the way it is. See, e.g. https://github.com/python/typing/issues/548#issuecomment-683336963